packages/core/src/artifacts.ts

```diff
@@ -181,17 +181,21 @@ export async function generateNodeModulesArtifacts(
   graphQLSchema: GraphQLSchema
 ) {
   const paths = getSystemPaths(cwd, config);
-
+  const dataProxy = config.db.url.startsWith('prisma:');
+  if (dataProxy === true) {
+    console.log('✨ Generating Prisma Client (data proxy)');
+  }
   await Promise.all([
-    generatePrismaClient(paths.schema.prisma),
+    generatePrismaClient(paths.schema.prisma, dataProxy),
     generateNodeModulesArtifactsWithoutPrismaClient(cwd, config, graphQLSchema),
   ]);
 }
 
-async function generatePrismaClient(prismaSchemaPath: string) {
+async function generatePrismaClient(prismaSchemaPath: string, dataProxyFlag: boolean) {
+  const dataProxy = dataProxyFlag || process.env.PRISMA_GENERATE_DATAPROXY === 'true';
```

Comment:
This env variable aligns with the prisma docs to generate for the data-proxy https://www.prisma.io/docs/data-platform/data-proxy/deploy

Reply:
I don't think we need that variable, it will only result in confusion about how it interacts with `"db.url": "prisma://"`

Reply:
@dcousens  this env var and cli flag is actually required for build envs where we have no access to the production environment variables.

E.g. docker builds in CI that prepare the code for production

Reply:
adding a `DATABASE_URL` which has `prisma://` in it just for the resulting prisma generate to build with proxy support feels kind of weird to me

Reply:
hmmm...so maybe I'm the only one to whom this kind of feels weird 😅 the postinstall/build/start commands where always a pain to me when doing CI stuff. It is really hard (without reading the code) what each of the commands does.

Most of the time, I found my self diving into the cli implementations of keystone, because we found weird behaviour when executing `keystone ...` commands (but this probably is something for a discussion thread)

Reply:
Please don't hold back with the feedback, it's really valuable! :yellow_heart: 
Adding `--data-proxy` is effectively free, but, the error messages seemed confusing if you accidentally mismatch.